react/components/EstimateShipping.tsx

```diff
@@ -4,39 +4,52 @@ import ShippingResult from './ShippingResult'
 import { useRuntime } from 'vtex.render-runtime'
 import { components, helpers } from 'vtex.address-form'
 import { newAddress } from '../utils/address'
+
 const { AddressContainer, AddressRules, StyleguideInput } = components
-const { addValidation } = helpers
+const { addValidation, removeValidation } = helpers
 
-type CustomProps = {
+interface CustomProps {
+  insertAddress: any
   selectedAddress: Address
   deliveryOptions: DeliveryOption[]
   countries: string[]
+  selectDeliveryOption: any
 }
 
 const EstimateShipping: FunctionComponent<CustomProps> = ({
+  insertAddress,
   selectedAddress,
   deliveryOptions,
   countries,
+  selectDeliveryOption,
 }) => {
   const { account, culture } = useRuntime()
 
   const [address, setAddress] = useState<AddressWithValidation>(
-    addValidation(selectedAddress || newAddress({ country: culture.country }))
+    addValidation(
+      selectedAddress
+        ? newAddress(selectedAddress)
+        : newAddress({ country: culture.country })
+    )
   )
 
-  const [showResult, setShowResult] = useState<boolean>(false)
+  const [showResult, setShowResult] = useState<boolean>(
+    selectedAddress && selectedAddress.postalCode ? true : false
```

Comment:
Not a change request, just a comment, I guess you could've used the `!!` notation instead of ternary.

`(!!selectedAddress && selectedAddress.postalCode)`

Reply:
I'm gonna try it.